internal/core/application/trade_service.go

```diff
@@ -162,18 +170,24 @@ func (t *tradeService) GetMarketPrice(
 		return nil, err
 	}
 
-	price, previewAmount, err := getPriceAndPreviewForMarket(unspents, mkt, tradeType, amount)
+	price, previewAmount, err := getPriceAndPreviewForMarket(unspents, mkt, tradeType, amount, asset)
```

Comment:
shouldt we pass the fee to calculate proper preview

Reply:
we already include fees for both balanced and pluggable strategies.